NuRadioReco/framework/base_trace.py

```diff
@@ -77,13 +77,31 @@ def get_filtered_trace(self, passband, filter_type='butter', order=10, rp=None):
         spec *= filter_response
         return fft.freq2time(spec, self.get_sampling_rate())
 
-    def get_frequency_spectrum(self):
-        if self.__time_domain_up_to_date:
-            self._frequency_spectrum = fft.time2freq(self._time_trace, self._sampling_rate)
-            self._time_trace = None
-            # logger.debug("frequency spectrum has shape {}".format(self._frequency_spectrum.shape))
-            self.__time_domain_up_to_date = False
-        return np.copy(self._frequency_spectrum)
+    def get_frequency_spectrum(self, window=None):
+        """
+        Returns the frequency spectrum.
+
+        Parameters
+        ----------
+        window: array of bools (default: None)
+            If not None, specifies the time window to be used for the FFT. Has to have the same length as the trace.
```

Comment:
I'm not so sure about the term `window`, which to me implies e.g. Tukey/Hamming/etc. windows, none of which are supported here.  I suppose that's clear from the `array of bools` type specification, but maybe it's worth mentioning the word `mask` somewhere or even changing the name of the argument.